solutions/usaco-740.mdx

```diff
@@ -2,12 +2,154 @@
+<CPPSection>
+```cpp
+#include <bits/stdc++.h> 
+using namespace std;
+
+const int MAX_N = 20;
+int n;
+vector<vector<char>> image(MAX_N, vector<char>(MAX_N));
+vector<vector<bool>> visited(MAX_N, vector<bool>(MAX_N));
+int pcl_count = 0;
+
+// PCL described by top-left (i1,j1) and bottom-right (i2,j2) corner
+struct PCL {
+	int i1, j1, i2, j2;
+};
+
+// Floodfill to find connected regions
+int i_min, i_max, j_min, j_max;
+void floodfill(int i , int j, char color) {
+	if(
+		i < i_min || j < j_min || i > i_max || j > j_max ||
+		visited[i][j] ||
+		image[i][j] != color
+	) {
+		return;
+	}
+
+	visited[i][j] = true;
+
+	floodfill(i + 1, j, color);
+	floodfill(i - 1, j, color);
+	floodfill(i, j + 1, color);
+	floodfill(i, j - 1, color);
+}
+
+// Check whether a given region is a PCL
+bool is_pcl(int i1, int j1, int i2, int j2) {
+
+	// keep track of num of regions for each color A-Z
+	vector<int> region_count(26, 0);
+
+	// set bounds for floodfill
+	i_min = i1;
+	i_max = i2;
+	j_min = j1;
+	j_max = j2;
+
+	// Count # of connected regions for each color present in bounds
+	for (int i = i1; i <= i2; i++) {
+		for (int j = j1; j <= j2; j++) {
+			if (!visited[i][j]) {
+				char cur_color = image[i][j];
+				region_count[cur_color - 'A']++;
+				floodfill(i, j, cur_color);
+			}
+		}
+	}
+
+	// Reset visited vector for the next call
+	fill(visited.begin(), visited.end(), vector<bool>(MAX_N, false));
+
+	// Verify PCL conditions
+	int color_count = 0;
+	bool color_with_one_region = false;
+	bool color_with_more_regions = false;
+	for (int i = 0; i < 26; i++) {
+		if (region_count[i] != 0) {
+			color_count++;
+		}
+		if (region_count[i] == 1) {
+			color_with_one_region = true;
+		}
+		if (region_count[i] > 1) {
+			color_with_more_regions = true;
+		}
+	}
+
+	return (
+		color_count == 2 &&
+		color_with_one_region &&
+		color_with_more_regions
+	);
+}
+
+// Check if PCL a is inside PCL b
+bool is_inside(PCL a, PCL b) {
+	return (
+		a.i1 >= b.i1 &&
+		a.i2 <= b.i2 &&
+		a.j1 >= b.j1 &&
+		a.j2 <= b.j2
+	);
+}
+
+int main() {
+	freopen("where.in", "r", stdin);
+	freopen("where.out", "w", stdout);
+	cin >> n;
+	for (int i = 0; i < n; i++) {
+		for (int j = 0; j < n; j++) {
+			cin >> image[i][j];
+		}
+	}
+	vector<PCL> pcl_list;
+
+	// Brute force every rectangle to find PCLs
+	for (int i1 = 0; i1 < n; i1++) {
+		for (int j1 = 0; j1 < n; j1++) {
+			for (int i2 = 0; i2 < n; i2++) {
+				for (int j2 = 0; j2 < n; j2++) {
+					if (is_pcl(i1, j1, i2, j2)) {
+						PCL new_pcl = {i1, j1, i2, j2};
+						pcl_list.push_back(new_pcl);
+					}
+				}
+			}
+		}
+	}
+
+	// If a PCL is inside another PCL, don't count it
+	for (int i = 0; i < pcl_list.size(); i++) {
+		PCL current_pcl = pcl_list[i];
+		bool valid_pcl = true;
+		for (int j = 0; j < pcl_list.size(); j++) {
+			if (i == j) {continue;}
+			PCL other_pcl = pcl_list[j];
+			if (is_inside(current_pcl, other_pcl)) {
+				valid_pcl = false;
+				break;
+			}
+		}
+		if (valid_pcl) {
+			pcl_count++;
+		}
```

Comment:
```suggestion
			pcl_count += valid_pcl;
```

Reply:
Do you think we should leave it as 
```cpp
if (region_count[i] == 1) {
	color_with_one_region = true;
}
if (region_count[i] > 1) {
	color_with_more_regions = true;
}
```
Since it is more immediately apparent what it does (for me at least) than the shortened expression
`color_with_more_regions = color_with_more_regions || region_count[i] >= 1;`

Reply:
wrong segment of code lol

Reply:
yeah its not showing me a comment prompt underneath that one, idk why. Its grouping these together

Reply:
hm, yeah you can leave this as is (but don't resolve it), let's wait for a third opinion

Reply:
i think its okay either way

Reply:
harry is this about this suggestion or the one we're talking about

Reply:
both, but ig it could be more clear if u still add the if statements?

Reply:
thought about it, the first one should be just formatted as one if statement

Reply:
> thought about it, the first one should be just formatted as one if statement

Do you mean like this?
```cpp
if (region_count[i]  >= 1) {
	color_with_one_region = true;
        color_with_more_regions = true;
}
```
Because that wouldn't work since for the condition you need two colors : 
 - one color has exactly one region
 - another color has 2 or more regions
 
 so with this way, a color with 2 or more regions would set both flags to true which could falsely trigger the condition.

Reply:
OH
THE VARIABLES ARE DIFFERENT
SO SORRY LMAO